### PR TITLE
Expose additional stft arguments to spectrogram

### PR DESCRIPTION
**Pull request by faroit**
This addresses #770

### CODE REVIEW

File: torchaudio/functional.py [hunk 1]
```diff
@@ -52,9 +52,12 @@ def spectrogram(
...
         win_length: int,
         power: Optional[float],
-        normalized: bool
+        normalized: bool,
+        center: bool = True,
+        pad_mode: str = "reflect",
+        onesided: bool = True
 ) -> Tensor:
-    r"""Create a spectrogram or a batch of spectrograms from a raw audio signal.
+    r"""Create a spectrogram     or a batch of spectrograms from a raw audio signal.
```
Comment (vincentqb) [inline torchaudio/functional.py:60]: nit: intentional?

**Review by vincentqb — COMMENTED** [review]
Thanks! Could you add at least one test that covers these paths? I suggest adding another torchscript consistency test [here](https://github.com/pytorch/audio/blob/master/test/torchaudio_unittest/torchscript_consistency_impl.py#L24).

File: torchaudio/functional.py (hunk 1 above)
Comment (faroit) [reply torchaudio/functional.py]: mbp butterfly keyboard...

**Comment by faroit** [comment]
> Thanks! Could you add at least one test that covers these paths? I suggest adding another torchscript consistency test [here](https://github.com/pytorch/audio/blob/master/test/torchaudio_unittest/torchscript_consistency_impl.py#L24).

sure. will do. It would be nice if the master's branch would pass the unit tests...

**Comment by vincentqb** [comment]
> It would be nice if the master's branch would pass the unit tests...

Sorry about this -- the ones I see [here](https://github.com/pytorch/audio/runs/1104288264) look flaky to me. :\ Did you mean different ones? I've made a small edit to the pull request to refresh the tests and see if they are still happening.

**Comment by gibiansky** [comment]
Hi, this PR is really valuable and I hope it gets merged soon! I got bitten by the `center=True` default and was about to file a feature request when I found this.

**Comment by mthrok** [comment]
Yes, this is very useful PR and sorry for not being responsive enough. 
I have fixed most of the unit test setup.
Also I performed refactoring on `torchaudio/functiona.py`, so can you rebase? (now the `spectrogram` function live in `torchaudio/functional/functiona.py`)

**Comment by mthrok** [comment]
Also not for this PR but I am wondering if we can better align the function signature and default values to `torch.stft`

**Comment by gibiansky** [comment]
(Thank you @mthrok !!! Just want to say I appreciate how responsive you've been in the two issues I've filed, it's amazing.)

**Review by vincentqb — APPROVED** [review]
LGTM, let's see how the tests come after rebasing.

**Comment by vincentqb** [comment]
> Also I performed refactoring on `torchaudio/functiona.py`, so can you rebase? (now the `spectrogram` function live in `torchaudio/functional/functiona.py`)

Rebased :)

**Pull request merged**